src/main/java/edu/ucsb/cs156/dining/controllers/ReviewsController.java

```diff
@@ -109,5 +136,66 @@ public Review postReview(
 
         return savedReview;
     }
+    
+    /**
+     * Update a single review
+     * 
+     * @param id        id of the review to update
+     * @param incoming  the new review
+     * @return the updated review object
+     */
+    @Operation(summary= "Update a single review")
+    @PreAuthorize("hasRole('ROLE_USER')")
+    @PutMapping("/reviewer")
+    public Review updateReview(
+            @Parameter(name="id") @RequestParam Long id,
+            @RequestBody @Valid Review incoming) {
+            
+        Review review = reviewsRepository.findById(id)
+                .orElseThrow(() -> new EntityNotFoundException(Review.class, id));
+
+        review.setStars(incoming.getStars());
+        review.setReviewText(incoming.getReviewText());
+        review.setStatus("Awaiting Moderation");
+        review.setModId(null);
+        review.setModComments(null);
+        review.setLastEditedDate(LocalDateTime.now().truncatedTo(ChronoUnit.SECONDS));
+
+        reviewsRepository.save(review);
+
+        return review;
+    }
+
+    /**
+     * Moderate a single review
+     * 
+     * @param id        id of the review to moderate
+     * @param incoming  the new review
+     * @return the updated review object
+     */
+    @Operation(summary= "Moderate a single review")
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
+    @PutMapping("/moderator")
+    public Review moderateReview(
+            @Parameter(name="id") @RequestParam Long id,
+            @RequestBody @Valid Review incoming) {
+
+
+        Review review = reviewsRepository.findById(id)
+                .orElseThrow(() -> new EntityNotFoundException(Review.class, id));
+        
+        if( !incoming.getStatus().matches("Approved|Awaiting Approval|Rejected") ) {
+            throw new IllegalArgumentException("Status must be 'Awaiting Moderation', 'Approved', or 'Rejected'");
+        } 
```

Comment:
Can we make the error message and the status consistent?  Even better if you can make that DRY.

Maybe something like: 
```java
ArrayList<String> legalStatusValues = new ArrayList<String>(
     Arrays.asList("Awaiting Moderation","Approved","Rejected");
```     

which you then use to check, as well as use in the error message.

Note that you could use:

```
"Status must be one of: " + legalStatusValues
```

and the toString() method would convert it nicely.

For the check, you could use:

```java
 if (!legalStatusValues.contains(incoming.getStatus()))

Reply:
fixed